src/ruby/puppetserver-lib/puppet/server/settings.rb

```diff
@@ -0,0 +1,52 @@
+require 'puppet/server/puppet_config'
+
+class Puppet::Server::Settings
+
+  # @param global_settings[Puppet::Settings] The puppet settings that all threads will spawn from
+  # @param puppet_config[Hash]
+  def initialize(global_settings:, puppet_config:)
+    @global_settings = global_settings
+    @local_settings = Puppet::ThreadLocal.new
+    @puppet_config = puppet_config
+  end
+
+  def method_missing(method, *args, &block)
```

Comment:
definitely a nitpick, and possible just a personal preference, so feel free to ignore, buuut I feel like this belongs at the bottom of the class? especially since it uses a method defined below, I had to scroll down and back up to see what this was doing.

Reply:
This is anecdotal, but I feel like I generally see `method_missing` defined early on in a class near the `initialize`, so that it's clear it is delegating a significant portion of method calls to some other entity.

Reply:
that's fair